extra/modules/confiant-ad-quality/src/main/java/org/prebid/server/hooks/modules/com/confiant/adquality/v1/ConfiantAdQualityBidResponsesScanHook.java

```diff
@@ -0,0 +1,121 @@
+
+    private static final String CODE = "confiant-ad-quality-bid-responses-scan-hook";
+
+    private final RedisClient redisClient;
+
+    private final RedisScanStateChecker redisScanStateChecker;
+
+    private final PrivacyEnforcementService privacyEnforcementService;
+
+    public ConfiantAdQualityBidResponsesScanHook(
+            RedisClient redisClient,
+            RedisScanStateChecker redisScanStateChecker,
+            PrivacyEnforcementService privacyEnforcementService) {
+        this.redisClient = redisClient;
+        this.redisScanStateChecker = redisScanStateChecker;
+        this.privacyEnforcementService = privacyEnforcementService;
+    }
+
+    @Override
+    public Future<InvocationResult<AllProcessedBidResponsesPayload>> call(
+            AllProcessedBidResponsesPayload allProcessedBidResponsesPayload,
+            AuctionInvocationContext auctionInvocationContext) {
+        final BidRequest bidRequest = getBidRequest(auctionInvocationContext);
+        final List<BidderResponse> responses = allProcessedBidResponsesPayload.bidResponses();
+        final boolean isScanDisabled = redisScanStateChecker.isScanDisabled();
+
+        if (isScanDisabled) {
+            return getResult(new BidsScanResult(OperationResult.empty()), auctionInvocationContext);
+        }
+
+        final Promise<InvocationResult<AllProcessedBidResponsesPayload>> processed = Promise.promise();
+
+        redisClient.submitBids(BidsMapper.bidResponsesToRedisBids(bidRequest, responses))
+                .onComplete(result -> processed
+                        .complete(getResult(result.result(), auctionInvocationContext).result()));
+
+        return processed.future();
+    }
+
+    private BidRequest getBidRequest(AuctionInvocationContext auctionInvocationContext) {
+        final AuctionContext auctionContext = auctionInvocationContext.auctionContext();
+        final BidRequest bidRequest = auctionContext.getBidRequest();
+        final ActivityCallPayload activityCallPayload = BidRequestActivityCallPayload.of(
+                ActivityCallPayloadImpl.of(ComponentType.GENERAL_MODULE, ConfiantAdQualityModule.CODE),
+                bidRequest);
+        final boolean disallowTransmitGeo = !auctionContext.getActivityInfrastructure()
+                .isAllowed(Activity.TRANSMIT_GEO, activityCallPayload);
+
+        final User maskedUser = privacyEnforcementService
+                .maskUserConsideringActivityRestrictions(bidRequest.getUser(), true, disallowTransmitGeo);
+        final Device maskedDevice = privacyEnforcementService
+                .maskDeviceConsideringActivityRestrictions(bidRequest.getDevice(), true, disallowTransmitGeo);
+
+        return bidRequest.toBuilder()
+                .user(maskedUser)
+                .device(maskedDevice)
+                .build();
+    }
+
+    private Future<InvocationResult<AllProcessedBidResponsesPayload>> getResult(
```

Comment:
More like `makeResult` or `toInvocationResult`, or smth like that. Also, return type should not be a `Future` as this method doesn't perform any long-running async tasks.